### PR TITLE
ci: parallelize docker RUN builds whenver possible

### PR DESCRIPTION
**Pull request by devjgm**
I used `${NCPU:-4}` (with a default value) in all the `demo-*` files. I used `${NCPU}` in the non-demo files, because its always set in our builds.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7060)

### CODE REVIEW

File: ci/cloudbuild/dockerfiles/ubuntu-bionic-install.Dockerfile [hunk 1]
```diff
@@ -104,7 +104,7 @@ RUN curl -sSL https://github.com/google/crc32c/archive/1.1.0.tar.gz | \
       -DCRC32C_BUILD_BENCHMARKS=OFF \
       -DCRC32C_USE_GLOG=OFF \
       -H. -Bcmake-out -GNinja && \
-    cmake --build cmake-out --target install && \
+    cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
```
Comment (coryan) [inline ci/cloudbuild/dockerfiles/ubuntu-bionic-install.Dockerfile:107]: I thought you wanted "${NCPU}" in non-demo files?

File: ci/cloudbuild/dockerfiles/ubuntu-xenial.Dockerfile [hunk 2]
```diff
@@ -65,7 +65,7 @@ RUN curl -sSL https://github.com/abseil/abseil-cpp/archive/20210324.2.tar.gz | \
       -DBUILD_SHARED_LIBS=yes \
       -DCMAKE_CXX_STANDARD=11 \
       -H. -Bcmake-out -GNinja && \
-    cmake --build cmake-out --target install && \
+    cmake --build cmake-out --target install -- -j ${NCPU:-4} && \
```
Comment (coryan) [inline ci/cloudbuild/dockerfiles/ubuntu-xenial.Dockerfile:68]: Ditto.

**Review by coryan — APPROVED** [review]
Some inconsistencies that you may want to fix.

File: ci/cloudbuild/dockerfiles/ubuntu-bionic-install.Dockerfile (hunk 1 above)
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/ubuntu-bionic-install.Dockerfile]: Oops. Fixed. Re-testing.

File: ci/cloudbuild/dockerfiles/ubuntu-xenial.Dockerfile (hunk 2 above)
Comment (devjgm) [reply ci/cloudbuild/dockerfiles/ubuntu-xenial.Dockerfile]: Fixed.

**Pull request merged**